src/icloudpd/base.py

```diff
@@ -1289,6 +1299,13 @@ def core(
                 photos_count = recent
                 photos_enumerator = itertools.islice(photos_enumerator, recent)
 
+            # Optional: Only download the x oldest photos.
+            if oldest is not None:
+                photos_count = oldest
+                total_photos = len(photos_enumerator)
+                start = total_photos - oldest
+                photos_enumerator = itertools.islice(photos_enumerator, start, total_photos)
```

Comment:
I am curious how that works on large collections in iCloud.com. I _think_ we are processing images as a stream and any filtering is done on a fly instead of memory.